### PR TITLE
feat: expose more properties in Controls

### PR DESCRIPTION
**Pull request by ndonkoHenri**
## Summary by Sourcery

Enhance the Flet core library by introducing new properties across multiple controls, including TextField, Text, Icon, PopupMenuButton, GestureDetector, Draggable, and NavigationBarDestination, to provide more customization and interaction options. Refactor existing functions and controls to improve flexibility, error handling, and support for adaptive design.

New Features:
- Introduce new properties to the TextField control, including cursor_error_color, obscuring_character, enable_interactive_selection, and more, enhancing customization options.
- Add new properties to the Text control, such as show_selection_cursor, enable_interactive_selection, and on_selection_change, to improve text interaction capabilities.
- Implement new properties for the Icon control, including shadows, fill, apply_text_scaling, grade, weight, and optical_size, allowing for more detailed icon customization.
- Introduce new properties to the PopupMenuButton control, such as menu_padding, style, popup_animation_style, and size_constraints, for enhanced menu customization.
- Add new properties to the GestureDetector control, including exclude_from_semantics, trackpad_scroll_causes_scale, and allowed_devices, to refine gesture handling.
- Introduce new properties to the Draggable control, such as axis, affinity, and max_simultaneous_drags, to enhance drag-and-drop functionality.
- Add new properties to the NavigationBarDestination control, including adaptive, to support adaptive design.
- Introduce new properties to the IconButton control, such as size_constraints, for better layout control.

Enhancements:
- Refactor the buildInputDecoration function to support additional parameters like prefixIcon and suffixIcon, improving input decoration flexibility.
- Enhance the parse functions for various controls to support default values and improve error handling.
- Refactor the MenuBar control to streamline the creation of child controls and improve code readability.
- Improve the handling of adaptive properties across various controls to better support platform-specific adaptations.

### CODE REVIEW

**Comment by FeodorFitsner** [comment]
Please resolve conflicts.

File: packages/flet/lib/src/controls/text.dart [hunk 1]
```diff
@@ -98,19 +108,63 @@ class TextControl extends StatelessWidget with FletStoreMixin {
       TextOverflow overflow =
           parseTextOverflow(control.attrString("overflow"), TextOverflow.clip)!;
 
+      onSelectionChanged(
+          TextSelection selection, SelectionChangedCause? cause) {
+        debugPrint("Markdown ${control.id} selection changed");
+        backend.triggerControlEvent(
```
Comment (FeodorFitsner) [inline packages/flet/lib/src/controls/text.dart:114]: Markdown?

File: sdk/python/packages/flet-core/src/flet_core/chip.py [hunk 2]
```diff
@@ -205,7 +205,7 @@ def before_update(self):
         self._set_attr_json("labelStyle", self.__label_style)
         self._set_attr_json("padding", self.__padding)
         self._set_attr_json("shape", self.__shape)
-        self._set_attr_json("color", self.__color)
+        self._set_attr_json("color", self.__color, wrap_attr_dict=True)
```
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/chip.py:208]: I see it's been changed in multiple controls. Do you have any examples/tests to see if anything is broken by that?

File: sdk/python/packages/flet-core/src/flet_core/types.py [hunk 3]
```diff
@@ -67,8 +66,6 @@ class UrlTarget(Enum):
 
 OffsetValue = Optional[Union[Offset, Tuple[Union[float, int], Union[float, int]]]]
 
-AnimationValue = Optional[Union[bool, int, Animation]]
```
[the remark this answers is not in the document]
Comment (ndonkoHenri) [reply sdk/python/packages/flet-core/src/flet_core/types.py:70]: That's right. Will remove the optional on `AnimationValue`.
Will look at the other `*Value`s we have in another PR.

**Review by FeodorFitsner — APPROVED** [review]

**Pull request merged**